experimental/AlgebraicStatistics/src/GaussianGraphicalModels.jl

```diff
@@ -0,0 +1,412 @@
+###################################################################################
+#
+#       Gaussian Rings
+#
+###################################################################################
+
+
+
+struct GaussianRing
+    ring::Ring
+    gens::Dict
+    covariance_matrix
+end
+
+
+
+@doc raw"""
+    gaussian_ring(n::Int; s_var_name::String="s", base_ring=QQ)
+
+A polynomial ring whose variables correspond to the entries of a covariance matrix of `n` Gaussian random variables.
+It is a multivariate polynomial ring whose variables are named `s[i,j]`and whose `base_ring` is by default `QQ`. 
+
+## Examples
+
+``` jldoctest gaussian_ring
+julia> R = gaussian_ring(3)
+Multivariate polynomial ring over Rational field in 6 variables
+s[1, 1], s[1, 2], s[1, 3], s[2, 2], s[2, 3], s[3, 3]
+```
+"""
+function gaussian_ring(n::Int, s_var_name::String="s", base_ring::Field=QQ)
+
+    S, s = polynomial_ring(base_ring, reduce(vcat, [[s_var_name*string([i,j]) for j in i:n] for i in 1:n]))
+
+    s = Dict([(Tuple(var_index(x)), x) for x in s])
```

Comment:
`s` changes its type here, from vector to dict, which makes the code type unstable. Just rename one. And while at it, rename `s_var_name`... ? How about this then
```suggestion
function gaussian_ring(n::Int, s::VarName="s", base_ring::Field=QQ)

    S, v = polynomial_ring(base_ring, reduce(vcat, [[s_var_name*string([i,j]) for j in i:n] for i in 1:n]))

    d = Dict([(Tuple(var_index(x)), x) for x in v)])
```
and then use `d` in the next two lines, too